packages/react-native/src/helpers.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect, useMemo } from "react";
+import { Animated, Easing } from "react-native";
+import { Themes } from "@pubnub/common-chat-components";
+
+export interface UseStyleProps<T> {
+  theme: Themes | "";
+  createDefaultStyle: (theme: Themes | "") => T;
+  customStyle?: T;
+}
+
+export const useStyle = <T>(props: UseStyleProps<T>): T => {
+  const { theme, createDefaultStyle, customStyle } = props;
+  const [style, setStyle] = useState(createDefaultStyle(theme));
+
+  useEffect(() => {
+    setStyle(Object.assign({}, createDefaultStyle(theme), customStyle));
+  }, [createDefaultStyle, setStyle, customStyle, theme]);
+
+  return style;
+};
+
+export const useRotation = (shouldRun = true): Animated.AnimatedInterpolation => {
```

Comment:
In the future I'd like to see that we use Reanimated v2. Reanimated is a very popular RN library for animations which has many advantages over RN's Animated module. It creates a new native thread and is completely independent from JS code execution. You can still "useNativeDriver" with RN's Animated module but it has limited support and doesn't support e.g. "top" property.